modular_skyrat/modules/novaya_ert/code/mod_suit.dm

```diff
@@ -0,0 +1,223 @@
+	reagents.add_reagent(reagent_required, reagent_max_amount)
+
+/obj/item/mod/module/auto_doc/on_activation()
+	. = ..()
+	if(!.)
+		return
+	RegisterSignal(mod.wearer, COMSIG_CARBON_HEALTH_UPDATE, .proc/on_use)
+	drain_power(use_power_cost)
+
+/obj/item/mod/module/auto_doc/on_use()
+	if(!COOLDOWN_FINISHED(src, heal_timer))
+		return FALSE
+	if(!check_power(use_power_cost))
+		balloon_alert(mod.wearer, "not enough charge!")
+		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
+		return FALSE
+	if(!allowed_in_phaseout && istype(mod.wearer.loc, /obj/effect/dummy/phased_mob))
+		to_chat(mod.wearer, span_warning("You cannot activate this right now."))
+		return FALSE
+	if(SEND_SIGNAL(src, COMSIG_MODULE_TRIGGERED) & MOD_ABORT_USE)
+		return FALSE
+	if(!reagents.has_reagent(reagent_required, reagent_required_amount))
+		balloon_alert(mod.wearer, "not enough chems!")
+		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
+		return FALSE
+	if(mod.wearer.health > health_threshold)
+		return
+	var/new_bruteloss = mod.wearer.getBruteLoss()
+	var/new_fireloss = mod.wearer.getFireLoss()
+	var/new_toxloss = mod.wearer.getToxLoss()
+	var/new_stamloss = mod.wearer.getStaminaLoss()
+	playsound(mod.wearer, 'modular_skyrat/modules/hev_suit/sound/hev/hiss.ogg', 100)
+	if(new_bruteloss)
+		mod.wearer.adjustBruteLoss(-heal_amount)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/sal_acid, 5)
+		to_chat(mod.wearer, span_warning("Brute treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))
+	if(new_fireloss)
+		mod.wearer.adjustFireLoss(-heal_amount)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/oxandrolone, 5)
+		to_chat(mod.wearer, span_warning("Burn treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))
+	if(new_toxloss)
+		mod.wearer.adjustToxLoss(-heal_amount)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/pen_acid, 5)
+		to_chat(mod.wearer, span_warning("Toxin treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))
+	if(new_stamloss)
+		mod.wearer.adjustStaminaLoss(-stamina_heal_amount)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 10)
+		to_chat(mod.wearer, span_warning("Combat stimulants administered. Overdose risks present on further use, consult your first-aid analyzer."))
+	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/coagulant, 5)
+	reagents.remove_reagent(reagent_required, reagent_required_amount)
+	drain_power(use_power_cost*10)
+	addtimer(CALLBACK(src, .proc/boost_aftereffects, mod.wearer), 45 SECONDS)
+	COOLDOWN_START(src, heal_timer, heal_cooldown)
```

Comment:
This is so hard to read, let it breathe.

I also feel like the stuff that keeps on getting repeated in those last few ifs should be made into a proc instead.

Also, why is it both _healing_ *and* _injecting reagents_? That feels a tad overpowered, and by a tad, I mean very very strong.
```suggestion
	if(!COOLDOWN_FINISHED(src, heal_timer))
		return FALSE

	if(!check_power(use_power_cost))
		balloon_alert(mod.wearer, "not enough charge!")
		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
		return FALSE

	if(!allowed_in_phaseout && istype(mod.wearer.loc, /obj/effect/dummy/phased_mob))
		to_chat(mod.wearer, span_warning("You cannot activate this right now."))
		return FALSE

	if(SEND_SIGNAL(src, COMSIG_MODULE_TRIGGERED) & MOD_ABORT_USE)
		return FALSE

	if(!reagents.has_reagent(reagent_required, reagent_required_amount))
		balloon_alert(mod.wearer, "not enough chems!")
		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
		return FALSE

	if(mod.wearer.health > health_threshold)
		return

	var/new_bruteloss = mod.wearer.getBruteLoss()
	var/new_fireloss = mod.wearer.getFireLoss()
	var/new_toxloss = mod.wearer.getToxLoss()
	var/new_stamloss = mod.wearer.getStaminaLoss()
	playsound(mod.wearer, 'modular_skyrat/modules/hev_suit/sound/hev/hiss.ogg', 100)

	if(new_bruteloss)
		mod.wearer.adjustBruteLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/sal_acid, 5)
		to_chat(mod.wearer, span_warning("Brute treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_fireloss)
		mod.wearer.adjustFireLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/oxandrolone, 5)
		to_chat(mod.wearer, span_warning("Burn treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_toxloss)
		mod.wearer.adjustToxLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/pen_acid, 5)
		to_chat(mod.wearer, span_warning("Toxin treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_stamloss)
		mod.wearer.adjustStaminaLoss(-stamina_heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 10)
		to_chat(mod.wearer, span_warning("Combat stimulants administered. Overdose risks present on further use, consult your first-aid analyzer."))

	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/coagulant, 5)
	reagents.remove_reagent(reagent_required, reagent_required_amount)
	drain_power(use_power_cost*10)
	addtimer(CALLBACK(src, .proc/boost_aftereffects, mod.wearer), 45 SECONDS)
	COOLDOWN_START(src, heal_timer, heal_cooldown)
```

Reply:
I've designed it with only the ERT in mind, and thus didn't care *that* much about the balancing. I'll remove the damage healing part; leaving chemicals.